src/ring_theory/integral_closure.lean

```diff
@@ -281,3 +282,87 @@ begin
   replace hmp := congr_arg subtype.val hmp,
   exact subtype.eq hmp
 end
+
+end
+
+section algebra
+open algebra
+variables {R : Type*} {A : Type*} {B : Type*}
+variables [comm_ring R] [comm_ring A] [comm_ring B]
+variables [algebra R A] [algebra A B]
+variables (A_alg : ∀ x : A, is_integral R x)
+
+include A_alg
+
+set_option class.instance_max_depth 50
+
+lemma is_integral_trans_aux (x : B) {p : polynomial A} (pmonic : monic p) (hp : aeval A B x p = 0)
+  (S : set (comap R A B))
+  (hS : S = (↑((finset.range (p.nat_degree + 1)).image
+    (λ i, to_comap R A B (p.coeff i))) : set (comap R A B))) :
+  is_integral (adjoin R S) (comap.to_comap R A B x) :=
```

Comment:
By the way, this lemma is basically trivial and doesn't use `A_alg` right?

Reply:
Yep, I've moved the include down a bit. And indeed, this lemma is trivial. That's why it has the longest proof (-;